jax_galsim/moffat.py

```diff
@@ -0,0 +1,444 @@
+                self._r0 = MoffatCalculateSRFromHLR(self._hlr, trunc, beta)
+
+            self._fwhm = self._r0 * (2.0 * jnp.sqrt(2.0 ** (1.0 / beta) - 1.0))
+
+        elif fwhm is not None:
+            if scale_radius is not None:
+                raise _galsim.GalSimIncompatibleValuesError(
+                    "Only one of scale_radius, half_light_radius, or fwhm may be specified",
+                    half_light_radius=half_light_radius,
+                    scale_radius=scale_radius,
+                    fwhm=fwhm,
+                )
+
+            self._fwhm = fwhm
+            self._r0 = self._fwhm / (2.0 * jnp.sqrt(2.0 ** (1.0 / beta) - 1.0))
+
+        elif scale_radius is not None:
+            self._r0 = scale_radius
+            self._fwhm = self._r0 * (2.0 * jnp.sqrt(2.0 ** (1.0 / beta) - 1.0))
+        else:
+            raise _galsim.GalSimIncompatibleValuesError(
+                "One of scale_radius, half_light_radius, or fwhm must be specified",
+                half_light_radius=half_light_radius,
+                scale_radius=scale_radius,
+                fwhm=fwhm,
+            )
+
+        # finalise the init HLR/ScaleRadius/FWHM determination
+        self._inv_r0 = 1.0 / self._r0
+
+        # maxRrD = maxR/rd ; fluxFactor Integral of total flux in terms of 'rD' units.
+        if trunc > 0.0:
+            _maxRrD = trunc * self._inv_r0
+            _fluxFactor = 1.0 - jnp.power(1 + _maxRrD * _maxRrD, (1.0 - beta))
+        else:
+            _fluxFactor = 1.0
+            ## Set maxRrD to the radius where missing fractional flux is xvalue_accuracy
+            ## (1+(R/rd)^2)^(1-beta) = xvalue_accuracy
+            ## JEC: notice this is w/o the Moffat normalisation
+            _maxRrD = jnp.sqrt(
+                jnp.power(gsparams.xvalue_accuracy, 1.0 / (1.0 - beta)) - 1.0
+            )
+
+        if half_light_radius is None:
+            self._hlr = self._r0 * jnp.sqrt(
+                jnp.power(1.0 - 0.5 * _fluxFactor, 1.0 / (1.0 - self._beta)) - 1.0
+            )
+
+        # Some other variables
+        _maxR = _maxRrD * self._r0  # maximum r
+        _maxR_sq = _maxR * _maxR
+        self._maxRrD_sq = _maxRrD * _maxRrD
+
+        self._norm = (
+            flux * (beta - 1) / (jnp.pi * _fluxFactor * self._r0**2)
+        )  # Normalisation f(x) (trunc=0)
+        self._knorm = flux  # Normalisation f(k) (trunc = 0, k=0)
+        self._knorm_bis = (
+            flux
+            * 4.0
+            / (jnp.power(2.0, self._beta) * jnp.exp(jax.lax.lgamma(self._beta - 1.0)))
+        )  # Normalisation f(k) (trunc = 0; k=/= 0)
+
+        # Register some parameters (see if all are needed ?)
+        super().__init__(
+            beta=beta,
+            scale_radius=self._r0,
+            half_light_radius=self._hlr,
+            fwhm=self._fwhm,
+            trunc=trunc,
+            flux=flux,
+            gsparams=gsparams,
+        )
+
+        # determination of maxK
+        if trunc == 0:
```

Comment:
there is a lot of conditionals in the code below, will this be ok when jitting?

Reply:
Good question. I must develop a test along the line `def test_gaussian_jitting():` in `test_jitting.py` to properly answer.